modules/apps/27-interchain-accounts/controller/keeper/account.go

```diff
@@ -41,7 +41,48 @@ func (k Keeper) RegisterInterchainAccount(ctx sdk.Context, connectionID, owner,
 
 	k.SetMiddlewareEnabled(ctx, portID, connectionID)
 
-	_, err = k.registerInterchainAccount(ctx, connectionID, portID, version, channeltypes.ORDERED)
+	_, err = k.registerInterchainAccount(ctx, connectionID, portID, version, channeltypes.UNORDERED)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// RegisterInterchainAccountWithOrdering is the entry point to registering an interchain account:
+// - It generates a new port identifier using the provided owner string, binds to the port identifier and claims the associated capability.
+// - Callers are expected to provide the appropriate application version string.
+// - For example, this could be an ICS27 encoded metadata type or an ICS29 encoded metadata type with a nested application version.
+// - A new MsgChannelOpenInit is routed through the MsgServiceRouter, executing the OnOpenChanInit callback stack as configured.
+// - An error is returned if the port identifier is already in use. Gaining access to interchain accounts whose channels
+// have closed cannot be done with this function. A regular MsgChannelOpenInit must be used.
+//
+// Deprecated: this is a legacy API that is only intended to function correctly in workflows where an underlying authentication application has been set.
+// Calling this API will result in all packet callbacks being routed to the underlying application.
+
+// Please use MsgRegisterInterchainAccount for use cases which do not need to route to an underlying application.
+
+// Prior to v6.x.x of ibc-go, the controller module was only functional as middleware, with authentication performed
+// by the underlying application. For a full summary of the changes in v6.x.x, please see ADR009.
+// This API will be removed in later releases.
+func (k Keeper) RegisterInterchainAccountWithOrdering(ctx sdk.Context, connectionID, owner, version string, ordering channeltypes.Order) error {
```

Comment:
I added this function to allow setting the ordering, otherwise if the controller chain has a a custom auth module and tries to open a channel with a host chain that doesn't support unordered channels, any attempt to open new ICA channels will fail. Using this function, the controller chain has the option to use ordered ordering with chains that don't support unordered ordering.

Reply:
I will open an issue to break the API of `RegisterInterchainAccount` to add the extra order parameter and also use `UNORDERED` ordering if `NONE` is passed.